subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/serialization/beans/BeanPropertyWriter.kt

```diff
@@ -47,6 +48,7 @@ class BeanPropertyWriter(
         for (field in relevantFields) {
             val fieldName = field.name
             val fieldValue = valueOrConvention(field.get(bean), bean, fieldName)
+            reportFieldProblems(fieldName, field.type, fieldValue)
```

Comment:
❓ Maybe report only once per field per type by hooking this check into the `relevantFields` computation?

Reply:
Here or in `relevantFields`, this would require keeping already seen bean types.
And we want the trace to be reported on each occurrence of the problem.
Problem deduplication in the reported summary should be enough.

Reply:
Implemented in https://github.com/gradle/gradle/pull/12760/commits/8268481037fcf8e4587229a0c9eb8a40f49d0a5e